src/hw/raspi4/startup.s

```diff
@@ -0,0 +1,97 @@
+.section ".text.boot"
+
+.globl _Reset
+_Reset:
+  // read cpu id, stop slave cores
+  mrs x1, mpidr_el1
+  and x1, x1, #3
+  cbz x1, 2f
+  // cpu id > 0, stop
+1:wfe
+  b 1b
+2:// cpu id == 0
+  
+  // set stack before our code
+  ldr x1, =stack_top
+  
+  // set up EL1
+  mrs x0, CurrentEL
+  and x0, x0, #12 // clear reserved bits
+  
+  // running at EL3?
+  cmp x0, #12
+  bne 5f
+  // should never be executed, just for completeness
```

Comment:
Explain why we don't expect to see this, and what the code below *would* do if it were to happen.